src/pages/en/guides/client-side-scripts.md

```diff
@@ -0,0 +1,211 @@
+  // Local imports are not resolved and will not work.
+</script>
+```
+
+:::note
+Adding `type="module"` or any attribute other than `src` to a `<script>` tag will disable Astro's default bundling behavior, treating the tag as if it had an `is:inline` directive.
+:::
+
+📚 See our [directives reference](/en/reference/directives-reference/#script--style-directives) page for more information about the directives available on `<script>` tags.
+
+### Script loading
+
+You may want to write your scripts as separate `.js`/`.ts` files or need to reference an external script on another server. You can do this by referencing these in a `<script>` tag’s `src` attribute.
+
+#### Import local scripts
+
+**When to use this:** If your script lives inside of `src/`.
+
+Astro will build, optimize, and add these scripts to the page for you, following its [script bundling rules](#script-bundling).
+
+```astro title="src/components/LocalScripts.astro"
+<!-- relative path to script at `src/scripts/local.js` -->
+<script src="../scripts/local.js"></script>
+
+<!-- also works for local TypeScript files -->
+<script src="./script-with-types.ts"></script>
+```
+
+#### Load external scripts
+
+**When to use this:** If your JavaScript file lives inside of `public/` or on a CDN.
+
+To load scripts outside of your project's `src/` folder, include the `is:inline` directive. This approach skips the JavaScript processing, bundling, and optimizations that are provided by Astro when you import scripts as described above.
+
+```astro title="src/components/ExternalScripts.astro" "is:inline"
+<!-- absolute path to a script at `public/my-script.js` -->
+<script is:inline src="/my-script.js"></script>
+
+<!-- full URL to a script on a remote server -->
+<script is:inline src="https://my-analytics.com/script.js"></script>
+```
+
+## Common script patterns
+
+### Handle `onclick` and other events
+
+Some UI frameworks use custom syntax for event handling like `onClick={...}` (React/Preact) or `@click="..."` (Vue). Astro follows standard HTML more closely and does not use custom syntax for events.
+
+Instead, you can use [`addEventListener`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) in a `<script>` tag to handle user interactions.
+
+```astro title="src/components/AlertButton.astro"
+<button class="alert">Click me!</button>
+
+<script>
+  // Find all buttons with the `alert` class on the page.
+  const buttons = document.querySelectorAll('button.alert');
+
+  // Handle clicks on each button.
+  buttons.forEach((button) => {
+    button.addEventListener('click', () => {
+      alert('Button was clicked!');
+    });
+  });
+</script>
+```
+
+:::note
+If you have multiple `<AlertButton />` components on a page, Astro will not run the component's script multiple times. Scripts are bundled and run only once. Using `querySelectorAll` ensures that this script attaches the event listener to every button with the `alert` class found on the page.
```

Comment:
Slight rework to try to address Kevin’s comment.

```suggestion
If you have multiple `<AlertButton />` components on a page, Astro will not run the script multiple times. Scripts are bundled and only included once per page. Using `querySelectorAll` ensures that this script attaches the event listener to every button with the `alert` class found on the page.
```

Reply:
Approve!